### ISSUE TITLE
Feature request: Allow custom icons/images for packages

### ISSUE THREAD

**Issue by indiejames** [atom/settings-view#438]
Currently packages use the authors' Github avatars for the icon/image in the package list. It would be better if the package author could specify an image.

**Comment by thedaniel** [comment]
Thanks for the feature request - this isn't a high priority (and I'm :-1: by default on anything that adds interface elements until persuaded otherwise) but I'll leave this issue open for discussion. 

For what it's worth, if we *did* do this I think that the best experience would probably involve adding something nice on atom.io for image picking and cropping rather than adding another `package.json` field (we want to be conservative about adding fields there).

**Comment by indiejames** [comment]
I would probably lean toward something simpler like having a prescribed file name/format/resolution at the top of the project. Something like icon.png (128x128) or similar.

**Comment by simurai** [comment]
I wonder what would be shown if a package wouldn't add a custom icon? I guess there could be a generic "placeholder" icon? But that doesn't look good if there is a lot of them.

Maybe another option would be to generate some kind of unique placeholder icon. Just so it doesn't look all the same. Something like this:

![screen shot 2015-03-21 at 5 45 50 pm](https://cloud.githubusercontent.com/assets/378023/6764418/2a17c6c2-cff2-11e4-97aa-481cad0e9d78.png)

Because thinking about it, I have a feeling that for most packages, it won't be easy to find a good icon and not every package author has some graphic design skills to create their own custom one.

**Comment by izuzak** [comment]
> Maybe another option would be to generate some kind of unique placeholder icon. Just so it doesn't look all the same. 

Great idea :+1:, something like identicons maybe? (but different): https://github.com/blog/1586-identicons

**Comment by simurai** [comment]
Right, like the identicons. Well, what I actually wanted to say: We should only consider adding package icons if we're confident that most packages will add a custom icon. The generated placeholders should only be as fallback for a few exceptions, not for the majority.

**Comment by braver** [comment]
I dunno, I really like the current method of using the avatars. It really humanises the effort that went into every package: *someone* made this. De-humazing it with generated nonsense like the identicons and google-style-letters to me really clashes with the Github community feel. It just ends up as a list of colour splotches that don't tell you anything. Instead just look at this discussion, or the list of packages you've got installed, or package lists on atom.io: there are all these *people* creating and contributing stuff. I'd add more avatars (like for all contributors of a package) rather taking any away.

So I'd say definitely default to the avatars. Now, if some package would define an icon in the package.json or something, you could take that. I could think of something cool for the Zen package for instance. But it shouldn't be the primary mode and I can definitely agree with @thedaniel on making it low priority: it's pretty awesome as is.

**Comment by simurai** [comment]
> It really humanises the effort that went into every package

:+1: I was more thinking in addition and not replacing the current avatars.. something like:

![package-icon](https://cloud.githubusercontent.com/assets/378023/6792104/273e711e-d1f8-11e4-8a6e-c4da797f6d87.png)

But I have a feeling that most packages won't add a custom icon and it will look less nice with a lot of "placeholder" or generated icons.

> Now, if some package would define an icon in the package.json or something, you could take that.

That would be another option, but it might also be strange.. some having "graphical icons" and others with human faces.

ok.. I agree keeping it as low priority.

**Comment by braver** [comment]
Oh right, that makes sense, sorry for not getting that. Still though, you're right: unless everyone makes awesome icons this just dilutes everything.

**Comment by abe33** [comment]
> It really humanises the effort that went into every package: someone made this. De-humazing it with generated nonsense like the identicons and google-style-letters to me really clashes with the Github community feel

> But I have a feeling that most packages won't add a custom icon and it will look less nice with a lot of "placeholder" or generated icons.

Same feelings here, If really a package needs a different visual identity than its primary author, then maybe he can use Github organization as a workaround (like it's already done with `atom-minimap` and `atom-community` orgs).
